tests/test_context.py

```diff
@@ -208,43 +208,41 @@ def test_parent_args_available_in_slots(self):
 
 class ContextCalledOnceTests(SimpleTestCase):
     def test_one_context_call_with_simple_component(self):
-        template = Template("{% load component_tags %}{% component_dependencies %}"
-                            "{% component name='incrementer' %}")
+        template = Template("{% load component_tags %}{% component name='incrementer' %}")
         rendered = template.render(Context()).strip()
 
         self.assertEqual(rendered, '<p class="incrementer">value=1;calls=1</p>', rendered)
```

Comment:
That these tests broke should mean that this is not backwards compatible right? Why did this test break?

Reply:
I thought this was automatically resolved by the new approach, but it's not.  The issue is that the `{% component_dependencies %}` tag would render a placeholder that would be included in the final output (because the middleware isn't used.  I modified the tag to check whether the middleware is active, and then to use the old behavior if it isn't.